src/test/java/edu/hm/hafner/analysis/IssueBuilderTest.java

```diff
@@ -0,0 +1,192 @@
+package edu.hm.hafner.analysis;
+
+
+import org.junit.jupiter.api.Test;
+
+import edu.hm.hafner.analysis.assertions.IssueSoftAssertions;
+
+/**
+ * Testing Class for IssueBuilder and Issue.
+ *
+ * @author Mark Tripolt
+ */
+class IssueBuilderTest {
+
+    @Test
+    void createDefaultIssue() {
+        Issue issue = new IssueBuilder().build();
+        IssueSoftAssertions.assertIssueSoftly(softly -> {
+            softly.assertThat(issue)
+                    .hasFileName("-")
+                    .hasLineStart(0)
+                    .hasLineEnd(0)
+                    .hasColumnStart(0)
+                    .hasColumnEnd(0)
+                    .hasCategory("")
+                    .hasType("-")
+                    .hasPackageName("-")
+                    .hasPriority(Priority.NORMAL)
+                    .hasMessage("")
+                    .hasDescription("")
+                    .hasToString("-(0,0): -: : ");
+        });
+
+    }
+
+    @Test
+    void createIssue() {
+        Issue issue = new IssueBuilder()
+                .setFileName("anotherIssue")
+                .setLineStart(1)
+                .setLineEnd(3)
+                .setColumnStart(0)
+                .setColumnEnd(123)
+                .setCategory("really bad")
+                .setType("worst")
+                .setPackageName("issue")
+                .setPriority(Priority.HIGH)
+                .setMessage("Just another Issue")
+                .setDescription("just another bad issue")
+                .build();
+
+        IssueSoftAssertions.assertIssueSoftly(softly -> {
+            softly.assertThat(issue)
+                    .hasFileName("anotherIssue")
+                    .hasLineStart(1)
+                    .hasLineEnd(3)
+                    .hasColumnStart(0)
+                    .hasColumnEnd(123)
+                    .hasCategory("really bad")
+                    .hasType("worst")
+                    .hasPackageName("issue")
+                    .hasPriority(Priority.HIGH)
+                    .hasMessage("Just another Issue")
+                    .hasDescription("just another bad issue")
+                    .hasToString("anotherIssue(1,0): worst: really bad: Just another Issue");
```

Comment:
das ist recht anfällig für format Änderungen

Reply:
Mir ist nicht ganz klar, was Sie hier genau meinen.
Habe ich das Problem nicht immer, wenn ich einen toString irgendwie überprüfen möchte und anschließend die toString Ausgabe ändere?

Reply:
Ja, bei toString hat man das meistens. Deswegen bietet sich folgende Ansätze an:
1. toString gar nicht testen (i.A. ok, da toString selten im API benutzt wird, sondern nur zum Debugging)
2. toString wie bei Ihnen auf Exaktheit testen (dann muss meist bei toString Anpassungen auch ein Test angepasst werden)
3. toString auf die wichtigsten Bestandteile mit mehreren contains prüfen: contains(fileName).contains(message)
Generell kann man keine Aussage treffen, welcher der 3 Fälle der beste ist. In unserem Fall finde ich die 3 am besten, kann mit der 2 aber auch gut leben